rs_server/s3_storage_handler/s3_storage_handler.py

```diff
@@ -17,243 +24,719 @@
+        """
+        if self.s3_client is None or bucket is None or s3_obj is None:
+            print("Input error for deleting the file")
+            return False
+        try:
+            with self.s3_client_mutex:
+                self.logger.debug("{%s} | {%s}", bucket, s3_obj)
+                self.logger.info("Delete file s3://{%s}/{%s}", bucket, s3_obj)
+                self.s3_client.delete_object(Bucket=bucket, Key=s3_obj)
+        except ClientError as e:
+            tb = traceback.format_exc()
+            self.logger.error("Failed to delete file s3://{%s}/{%s}", bucket, s3_obj)
+            self.logger.error("Exception: {%s} | {%s}", e, tb)
+            return False
+        return True
+
+    # helper functions
+    # function to read the secrets from .s3cfg or aws credentials files
+    @staticmethod
+    def get_secrets(secrets, secret_file, logger=None):
+        """Read secrets from a specified file.
+
+        Args:
+            secrets (dict): Dictionary to store retrieved secrets.
+            secret_file (str): Path to the file containing secrets.
+            logger (Logger, optional): Logger instance for error logging.
+
+        Returns:
+            bool: True if secrets are successfully retrieved, False otherwise.
+        """
+        try:
+            dict_filled = 0
+            with open(secret_file, "r", encoding="utf-8") as aws_credentials_file:
+                lines = aws_credentials_file.readlines()
+                for line in lines:
+                    if dict_filled == len(secrets):
+                        break
+                    if secrets["s3endpoint"] is None and "host_bucket" in line:
+                        dict_filled += 1
+                        secrets["s3endpoint"] = line.strip().split("=")[1].strip()
+                    elif secrets["accesskey"] is None and "access_key" in line:
+                        dict_filled += 1
+                        secrets["accesskey"] = line.strip().split("=")[1].strip()
+                    elif secrets["secretkey"] is None and "secret_" in line and "_key" in line:
+                        dict_filled += 1
+                        secrets["secretkey"] = line.strip().split("=")[1].strip()
+        except (OSError, IndexError) as e:
+            if logger:
+                logger.error(f"Could not get the secrets, exception: {e}")
+            else:
+                print(f"Could not get the secrets, exception: {e}")
+            return False
+        if secrets["accesskey"] is None or secrets["secretkey"] is None:
+            return False
+        return True
+
+    # get the filename only from a full path to it
+    @staticmethod
+    def get_basename(input_path):
+        """Get the filename from a full path.
+
+        Args:
+            int_path (str): The full path.
+
+        Returns:
+            str: The filename.
+        """
+        path, filename = ntpath.split(input_path)
+        return filename or ntpath.basename(path)
```

Comment:
What is the use of this function instead of just using `os.path.basename(input_path)` ?

Reply:
I have had this function for a long time, carrying it with me everywhere. It takes care in case of input_path ends with /. (it's also crossover, win and linux). I would prefer to leave it like this. Just make a test with get_basename("/test/t1/") and os.path.basename("/test/t1/") . The result is different -> t1 with my func, vs empty string with what you are suggesting